### PR TITLE
README: add semgrep rules

### LINKED ISSUE #16: semgrep rules?
I wrote some simple semgrep rules to help catch common mistakes (failing to defer try.Handle et al) and misuses (try.E without a function-local handler). They have some exceptions that are peculiar to our codebase, but we'd like to share a general purpose rules file, I could extract one.

### PR DESCRIPTION
**Pull request by josharian**
Fixes #16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,8 +89,59 @@ See the [documentation][godev] for more information.
...
+
+```yaml
+rules:
+  - id: non-deferred-try-handle
+    patterns:
+      - pattern-either:
+          - pattern: try.Handle(...)
+          - pattern: try.HandleF(...)
+          - pattern: try.F(...)
+          - pattern: try.Recover(...)
```
Comment (dsnet) [inline README.md:104]: Wanna sort these alphabetically (i.e., `F`, `Handle`, `HandleF`, `Recover`)?

File: README.md [hunk 2]
```diff
@@ -89,8 +89,59 @@ See the [documentation][godev] for more information.
...
+          ...
+          defer try.Handle(...)
+          ...
+      - pattern-not-inside: |
+          ...
+          defer try.HandleF(...)
+          ...
+      - pattern-not-inside: |
+          ...
+          defer try.F(...)
```
Comment (dsnet) [inline README.md:131]: Ditto, move this above `Handle`?

**Review by dsnet — APPROVED** [review]

**Comment by dsnet** [comment]
Thanks!

**Comment by josharian** [comment]
Done. Assuming I can alphabetize correctly today. Been one of those days... :P

**Pull request merged**